src/agoradatatools/etl/utils.py

```diff
@@ -106,23 +105,40 @@ def standardize_values(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
-def rename_columns(df: pd.DataFrame, column_map: dict) -> pd.DataFrame:
-    """Takes in a dataframe and renames columns according to the mapping provided
+def rename_columns(
+    data: Union[pd.DataFrame, list, dict], column_map: dict
+) -> Union[pd.DataFrame, list, dict]:
+    """Takes in a dataframe, list of dictionaries, or dictionary and renames columns according to the mapping provided
 
     Args:
-        df (pd.DataFrame): DataFrame with columns to be renamed
+        data (pd.DataFrame, list, dict): DataFrame, list of dictionaries, or dictionary with columns to be renamed
         column_map (dict): Dictionary mapping original column names to new columns
 
     Returns:
-        pd.DataFrame: DataFrame with new columns names
+        pd.DataFrame, list, dict: DataFrame, list of dictionaries, or dictionary with new columns names
     """
+
+    def _rename_dict(d: dict, column_map: dict) -> None:
+        """
+        Rename keys in a dictionary in place.
+        """
+        for old_key, new_key in column_map.items():
+            if old_key in d:
+                d[new_key] = d.pop(old_key)
+
     try:
-        df.rename(columns=column_map, inplace=True)
+        if isinstance(data, list):
```

Comment:
One potential issue: This only checks if data is a `list`, _not_ that each item in the list is a `dict`. Possibly this check could be done inside the `for old_key, new_key...` loop in `_rename_dict` so each item gets checked. 

OR... does enforcing `d: dict` auto-check that?

Reply:
Got it! Looks like you changed it to `list[dict]`, so we're good here.